src/app/AuthenticationController.tsx

```diff
@@ -26,11 +26,11 @@ interface AuthenticationControllerReduxProps {
 
 type AuthenticationControllerProps = AuthenticationControllerReduxProps & {
   protectedAreaComponent: React.ReactNode;
-  publicAreaComponent: React.ReactNode;
+  publicAreaComponent: (isPostLoginPerforming: boolean, errorMsg?: string) => React.ReactNode;
 };
 
 interface AuthenticationControllerState {
-  stage: 'login' | 'post-login' | 'logged-in';
+  stage: 'login' | 'post-login' | 'logged-in' | 'logged-in-at-load';
```

Comment:
If you used an `enum` for stage you wouldn't need to assign strings to it below and then it would be typesafe.